### PR TITLE
Feat: Improve search time + custom search in debug page + search input enhancement

### PR DESCRIPTION
**Pull request by ptbrowne**
- feat: Split query in two for search, 1st search in name + description, then search in creator, publisher, theme
- feat: Ability to use a custom search in debug search page
- fix: When sharing a search URL, search input was not retrieved

improves for #683 
fix https://github.com/visualize-admin/visualization-tool/issues/699

### CODE REVIEW

File: app/configurator/components/dataset-browse.tsx [hunk 1]
```diff
@@ -327,7 +317,7 @@ export const SearchDatasetBox = ({
 
   const isSearching = search !== "";
 
-  const onToggleIncludeDrafts = useCallback(() => {
+  const onToggleIncludeDrafts = useEvent(async () => {
     setIncludeDrafts(!includeDrafts);
   }, [includeDrafts, setIncludeDrafts]);
```
Comment (bprusinowski) [inline app/configurator/components/dataset-browse.tsx]: Async?

File: app/rdf/query-search.ts [hunk 2]
```diff
@@ -82,14 +82,16 @@ const makeVisualizeFilter = (includeDrafts: boolean) => {
...
-  );
+  const enhancedQuery = rawQuery
+    .toLowerCase()
+    .split(" ")
+    // Filter out lowercase, small tokens
+    .filter((t) => t.length > 2 || t.toLowerCase() !== t)
+    // Wildcard Searches on each term
+    .map((t) => `${t}*`)
+    .join(" ");
+  console.log({ enhancedQuery });
```
Comment (bprusinowski) [inline app/rdf/query-search.ts:93]: 👓

**Review by bprusinowski — APPROVED** [review]
LGTM!

File: app/configurator/components/dataset-browse.tsx (hunk 1 above)
Comment (ptbrowne) [reply app/configurator/components/dataset-browse.tsx]: It is used after.

**Pull request merged**